MATLAB/sensor_reading.m

```diff
@@ -58,3 +55,57 @@
 
 end
 
+
+function [sun_vec, success] = update_sun_sensors(real_n, measured_n, sat2sun, eclipse)
+%update_sun_sensors simulates the behavior of the sun sensor array and sun
+%vector determination algorithm on the ADCS. All vectors are in the body
+%frame.
+%   real_n is a matrix containing the real normal vectors of the sun
+%       sensors.
+%   measured_n is a matrix containing the actual normal vectors of the sun
+%       sensors.
+%   sat2sun is a unit vector pointing from the satellite to the sun in the
+%       body frame.
+%   eclipse is true if the satellite is in eclipse and false otherwise.
+%   sun_vec is the measured sun vector in the body frame of the spacecraft.
+%   success is true if the returned sun vector is accurate and false
+%       otherwise.
+
+% Assume failure or eclipse
+success = 0;
+sun_vec = NaN(3,1);
+
+if (~eclipse)
+
+    % TODO : Revisit voltage measurement generation
+    [~, N] = size(real_n);
+    voltages = zeros(N, 1);
+    for i = 1:N
+        %magnitude of a normal is the inverse of the max voltage reading so
+        %dividing by the frobenius norm gives a voltage measure.
+        voltages(i) = sat2sun' * real_n(:, i) / norm(real_n(:, i), 'fro');
```

Comment:
dividing real_n(:,i) by its norm just normalizes real_n(:,i). So voltages is cos(theta) now.